server/src/main/java/io/crate/metadata/settings/session/SessionSettingRegistry.java

```diff
@@ -128,7 +130,28 @@ public SessionSettingRegistry(Set<SessionSettingProvider> sessionSettingProvider
                      s -> Boolean.toString(s.errorOnUnknownObjectKey()),
                      () -> String.valueOf(true),
                      "Raises or suppresses ObjectKeyUnknownException when querying nonexistent keys to dynamic objects.",
-                     DataTypes.BOOLEAN));
+                     DataTypes.BOOLEAN))
+            .put(DATE_STYLE_KEY,
+                 new SessionSetting<>(
+                     DATE_STYLE_KEY,
+                     objects -> {
+                         if (objects.length != 1) {
+                             throw new IllegalArgumentException(DATE_STYLE_KEY + " should have only one argument.");
+                         }
+                     },
+                     objects -> DataTypes.STRING.implicitCast(objects[0]),
+                     (s, v) -> {
+                         if (!v.toUpperCase(Locale.ENGLISH).equals(String.valueOf(PostgresWireProtocol.DATE_STYLE))) {
+                             throw new UnsupportedOperationException("\"" + DATE_STYLE_KEY + "\" cannot be changed.");
```

Comment:
What do you think of keeping this as warning?

Reply:
After some thinking I tend to just ignore `SET Datestyle` similar to other setting, as this could potentially be a breaking change and log the message